src/java.base/unix/native/libjli/java_md_solinux.h

```diff
@@ -60,4 +66,29 @@ static const char *user_dir     = "/java";
 #include <pthread.h>
 #endif
 
+#ifdef AIX
+#define ZLIBNX_PATH "/usr/opt/zlibNX/lib"
+
+#ifndef POWER_9
+#define POWER_9 0x20000 /* 9 class CPU */
+#define POWER_9_ANDUP (POWER_9|POWER_10_ANDUP)
+#define __power_9_andup() (_system_configuration.implementation & POWER_9_ANDUP)
+#endif
+
+#ifndef POWER_10
+#define POWER_10 0x40000 /* 10 class CPU */
+#define POWER_10_ANDUP (POWER_10)
+#define __power_10_andup() (_system_configuration.implementation & POWER_10_ANDUP)
+#endif
```

Comment:
You're not actually using `__power_10_andup` and `__power_9_andup` doesn't do what its name implies.
I suggest we mirror the test in OMR:
```
#ifndef POWER_9
#define POWER_9 0x20000 /* 9 class CPU */
#endif

#ifndef POWER_10
#define POWER_10 0x40000 /* 10 class CPU */
#endif

#define __power_9_andup() ((POWER_9  == _system_configuration.implementation) \
                        || (POWER_10 == _system_configuration.implementation))

#ifndef SC_NX_CAP
#define SC_NX_CAP 60
#endif

#ifndef NX_GZIP_PRESENT
#define NX_GZIP_PRESENT 0x00000001
#endif

#define __power_nx_gzip() (0 != ((long)getsystemcfg(SC_NX_CAP) & NX_GZIP_PRESENT))
```
Also note the explicit comparison with zero in `__power_nx_gzip`.

Reply:
Updated in 726b2bb